app/views/posts/show.html.erb

```diff
@@ -0,0 +1,46 @@
+<section id = "user-post" class="user-post">
+<%= render partial: 'shared/nav_link' %>
+
+  <div class="post user_post">
+    <div class="post-header">
+    
+      <h2>
+        Post
+        by <%= @user.name %>
+      </h2>
+      <h4>
+        Comments: <%= @post.comments_counter %>,
+        Likes: <%= @post.likes_counter %>
+      </h4>
+    </div>
+    <h3><%= @post.title %></h3>
+    <p><%= @post.text %></p>
+    <% if can? :destroy, @post %>
+  <%= button_to "Delete", user_post_path(user_id: @post.author_id, id: @post.id), method: :delete, data: { confirm: "Are you sure you want to delete this post?" }, class: "delete-btn" %>
+<% end %>
+  </div>
+    <br>
+<div class=" user-item">
+       <%= render partial: 'shared/add_comment', locals: { user: @user, show_link: false  } %>
+    </div>
+
+  <div class="create-like-Comment see-all-posts">
+  <%= button_to "Like Post",  user_post_likes_url(user_id: current_user.id, post_id: @post.id),method: :post, class: "create-like-link see-link"%> 
+  </div>
+  
+<% if @post.comments_counter > 0 %>
+  <div class="comments post">
+    <% @comments.each do |comment| %>
+      <li><%= comment.user.name %>:
+      <%= comment.text %>
+       <% if can? :destroy, comment %>
+  <%= button_to "Delete Comment", user_post_comment_path(user_id: @post.author_id, post_id: @post.id, id: comment.id), method: :delete, data: { confirm: "Are you sure you want to delete this comment?" }, class: "delete-btn" %>
```

Comment:
- [ Required ] I think it would be nice if you make it possible for a user with the `role` of `admin` to be able to `delete` any post and any `comment` on this page. Currently, when I log in as a user with the role of `admin` the `delete`  post button does not delete a post when I click on it.  The `delete` comment button on the `posts` show page also does not delete a comment when I click on it. 

> Please note that a user with the role of `admin` should be able to delete any `post` or `comment` on the blog application.

![Screenshot from 2023-09-07 23-22-55](https://github.com/Kel-nana/Blog-app/assets/16392046/c7bd6c28-8f14-4e0e-9b9b-2c1e57a725ae)